db/crud.go

```diff
@@ -1315,14 +1315,34 @@ func (db *DatabaseCollectionWithUser) PutExistingCurrentVersion(ctx context.Cont
 				newGeneration = prevGeneration
 			}
 		}
-		revTreeConflictChecked := false
-		var parent string
-		var currentRevIndex int
 
 		// We want skip conflict check for the following scenarios:
 		// 1. incoming doc is a tombstone and local doc is also a tombstone (for ISGR pull replications)
 		// 2. local doc has no hlv
 		allowConflictingTombstone := opts.ForceAllowConflictingTombstone && doc.IsDeleted()
+
+		// if local doc is a pre-upgraded mutation and the rev tree is conflicting with incoming, assign this local
+		// version an implicit hlv based on its rev tree ID + this nodes sourceID to allow HLV conflict resolution to occur
+		// between the two versions.
+		if doc.HLV == nil && doc.Cas != 0 && !allowConflictingTombstone {
+			_, _, conflictErr := db.revTreeConflictCheck(ctx, opts.RevTreeHistory, doc, opts.NewDoc.Deleted)
+			if conflictErr != nil {
```

Comment:
```suggestion
			if errors.Is(conflictErr, base.ErrDocumentConflict) {
```

you'd have to define `ErrDocumentConflict = base.HTTPErrorf(http.StatusConflict, "Document revision conflict")` but this would future proof this function if it returns a different type of error in the future, and it was not obvious to me that this function would only return this error type